File: backend/test-campaign.py
import os

import django

os.environ.setdefault("DJANGO_SETTINGS_MODULE", "core.settings")
django.setup()

from django.test import Client

from apps.authentication.models import User

client = Client()
owner = User.objects.get(email="test_owner@loyallia.com")
owner.set_password("123456")
owner.save()

login_res = client.post(
    "/api/v1/auth/login/",
    {"email": "test_owner@loyallia.com", "password": "123456"},
    content_type="application/json",
)
token = login_res.json()["access_token"]
headers = {"HTTP_AUTHORIZATION": f"Bearer {token}"}

create_res = client.post(
    "/api/v1/notifications/campaigns/",
    {"title": "TEST CAMPAIGN", "message": "Test", "segment_id": "all"},
    content_type="application/json",
    **headers,
)
print("CREATE:", create_res.status_code, create_res.json())

list_res = client.get("/api/v1/notifications/campaigns/", **headers)
print("LIST:", list_res.status_code, list_res.json())

File: frontend/test-campaign.js
const http = require('http');

const data = JSON.stringify({ email: "test_owner@loyallia.com", password: "123456" });

const req = http.request(
    { hostname: 'localhost', port: 8000, path: '/api/v1/auth/login/', method: 'POST', headers: {'Content-Type': 'application/json', 'Content-Length': data.length} },
    res => {
        let body = '';
        res.on('data', d => body += d);
        res.on('end', () => {
            const token = JSON.parse(body).access_token;
            
            const createReq = http.request(
                { hostname: 'localhost', port: 8000, path: '/api/v1/notifications/campaigns/', method: 'POST', headers: {'Authorization': `Bearer ${token}`, 'Content-Type': 'application/json'} },
                cRes => {
                    let cBody = '';
                    cRes.on('data', d => cBody += d);
                    cRes.on('end', () => {
                        console.log("CREATE:", cRes.statusCode, cBody);
                        
                        const listReq = http.request(
                            { hostname: 'localhost', port: 8000, path: '/api/v1/notifications/campaigns/', method: 'GET', headers: {'Authorization': `Bearer ${token}`} },
                            lRes => {
                                let lBody = '';
                                lRes.on('data', d => lBody += d);
                                lRes.on('end', () => console.log("LIST:", lRes.statusCode, lBody));
                            }
                        );
                        listReq.end();
                    });
                }
            );
            createReq.write(JSON.stringify({ title: "NodeJS Test", message: "Wow", segment_id: "all" }));
            createReq.end();
        });
    }
);
req.write(data);
req.end();
